src/provider/provider_tracking.c

```diff
@@ -404,3 +434,27 @@ void umfTrackingMemoryProviderGetUpstreamProvider(
         (umf_tracking_memory_provider_t *)hTrackingProvider;
     *hUpstream = p->hUpstream;
 }
+
+void umfTrackingMemoryProviderFini(void *tracker) {
+#ifndef NDEBUG
+    uintptr_t rkey;
```

Comment:
you repeat the same code - could you create a function for that?

Reply:
Done